staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -107,6 +107,15 @@ func MonitorRequest(request *http.Request, verb, resource, subresource, scope, c
 		}
 	}
 
+	// make get pod log with follow option api request reported as watch
+	if verb == "GET" && resource == "pods" && subresource == "log" {
```

Comment:
No kube-specific types are allowed in the *generic* apiserver.  This knowledge doesn't work generically.

Reporting should already be per-subresource (it can be anyway). What is going wrong with the accounting?